Update the README.

PeterJCLaw opened the pull request:
This makes it easier for first time users by removing the apparent
need to install development things (since vagrant is now in many
disto's repos). Also remove the comment that it's our new server,
since it's been around for quite a while.

PeterJCLaw committed:
Update the README.

This makes it easier for first time users by removing the apparent
need to install development things (since vagrant is now in many
disto's repos). Also remove the comment that it's our new server,
since it's been around for quite a while.

fables-tales commented on readme.md:
```diff
@@ -1,35 +1,37 @@
-is a vagrant configuration that will very easily go from scratch to a vagrantbox.
+Badger is the [Student Robotics](http://studentrobotics.org) server.
+This is a vagrant configuration that will very easily go from scratch
+to a VagrantBox.
```
This is jargon. Imagine I'm new, what's a vagrant box?

fables-tales commented on readme.md:
```diff
@@ -1,35 +1,37 @@
+  * Or, if you like installing from source, you can install Ruby (1.9.3+,
+    including development headers: `yum install ruby-devel libffi-devel` or
+    `apt-get install ruby-dev libffi-dev`) and then `gem install vagrant`
+3. `git clone https://github.com/samphippen/badger-vagrant && cd badger-vagrant`
```
I think all of these should have a `run` before them so that people know what they're doing.

fables-tales commented:
A few notes. Otherwise great.

PeterJCLaw committed:
Correct the completion message in build.sh

This was upside down, such that it would print "success" on error.

PeterJCLaw commented:
This pull request is actually likely to include a bunch of other related fixes -- not just the README.

PeterJCLaw committed:
Further changes to make the README easier.

Primarily aimed at new people -- provide some idea of what a 'vagrant
box' is, add the instruction to run the commands in backtick quotes.

PeterJCLaw committed:
Also detect and error about puppet fails.

This is more likely than any other failure, but it's unclear if we
don't error about it upfront.

PeterJCLaw committed:
Even more detail about how to access the VM.

Not everyone knows what NAT means, or how to use the port information.
Provide this via some trivial links and more info about sshing in.

PeterJCLaw committed:
Force-link the localhost links when rendering.

This makes them easily clickable in a browser, thus making it even
easier to browse to the VM's pages.

Change-Id: I79d9d940b5da3c89afab21c884af44debf002799

fables-tales commented:
@PeterJCLaw how done is this should I merge it?

PeterJCLaw commented:
I think it's done now, I've not got any other changes planned. Not sure the title of this PR is quite right though -- might want a tweak if you do merge it.

fables-tales commented:
I'm just gonna merge it now.
